### PR TITLE
Eslint deps docs

### PR DESCRIPTION
**Pull request by coffee-cup**
- fix deps of eslint-plugin
- update readme for eslint-plugin
- make eslint-plugin public

### CODE REVIEW

File: packages/eslint-plugin/README.md [hunk 1]
```diff
@@ -7,33 +7,45 @@
 You'll first need to install [ESLint](http://eslint.org):
```
Comment (tdawes) [inline packages/eslint-plugin/README.md]:
```suggestion
First, you'll need to install [ESLint](http://eslint.org):
```

File: packages/eslint-plugin/README.md [hunk 2]
```diff
@@ -7,33 +7,45 @@
...
 
 ```
-$ npm i eslint --save-dev
+$ npm install eslint --save-dev
 ```
 
-Next, install `@prodo/eslint-plugin`:
+Next install the typescript eslint parser:
```
Comment (tdawes) [inline packages/eslint-plugin/README.md]:
```suggestion
Next, install the typescript eslint parser:
```

File: packages/eslint-plugin/README.md [hunk 3]
```diff
@@ -4,36 +4,48 @@
...
 
-**Note:** If you installed ESLint globally (using the `-g` flag) then you must also install `@prodo/eslint-plugin` globally.
+**Note:** If you installed ESLint globally (using the `-g` flag) then you must
+also install `@prodo/eslint-plugin` globally.
 
 ## Usage
 
-Add `@prodo/eslint-plugin` to the plugins section of your `.eslintrc` configuration file.prefix:
+Add `@prodo/eslint-plugin` to the plugins section of your `.eslintrc`
+configuration, and set the parser option to `@typescript-eslint/parser`:
```
Comment (tdawes) [inline packages/eslint-plugin/README.md]: What if they're only using JS?

File: packages/eslint-plugin/README.md [hunk 4]
```diff
@@ -4,36 +4,48 @@
...
+It is recomended that you also use the [typescript-eslint
+plugin](https://github.com/typescript-eslint/typescript-eslint).
+
 Then configure the rules you want to use under the rules section.
 
 ```json
 {
   "rules": {
-    "@prodo/eslint-plugin/rule-name": 2
+    "@prodo/rule-name": 2
```
Comment (tdawes) [inline packages/eslint-plugin/README.md]: Maybe provide an actual example of the available rules?

**Comment by coffee-cup** [comment]
@tom Moved the docs in the readme to the actual docs.

Also, currently it does not work unless you are using the `@typescript/eslint-parser`. I left a note about that in the docs.

**Comment by kprimerakis** [comment]
To clarify, it does work without the typescript parser, so perhaps we can treat typescript the same way we treat react? Start with the basic: install eslint. If typescript, install typescript eslint and set parser. If react... and so on.

**Comment by coffee-cup** [comment]
@kprimerakis have a look at the latest commit and see if that works

**Comment by kprimerakis** [comment]
Looks sensible to me.

**Pull request merged**